### DISCUSSION TITLE
Woche 8 - Utility Tree und Architekturentscheidungen

### DISCUSSION THREAD

**Discussion by kattyterra** [Chrissi-Ruege/HealthKeeper#20] [Q&A]
Health Keeper als Multipage-Webanwendung wurde entworfen, um ein breites Spektrum an Gesundheits- und Fitnessfunktionen zu bieten. Die Architektur folgt einem Client-Server-Modell, wobei ASP.NET als Grundlage dient. Dies ermöglicht die Integration verschiedener Technologien wie Entity Framework Core, JavaScript sowie HTML/CSS für ein robustes Frontend.
Die Anwendung ist modular gestaltet, wobei jede Funktion über eigene Controller navigiert wird. Diese beinhalten Grafiken zur Gewichtsverfolgung, einen BMI Rechner, ein Ernährungstagebuch, eine visuelle Darstellung des Wasserkonsums, einen Aktivitätentracker für Sport, sowie einen Kalender für Trainingspläne und personalisierte Gesundheits- und Fitnessvideos. Health Keeper umfasst auch eine Konto-Funktionalität, die es den Benutzern ermöglicht, personalisierte Konten zu erstellen und sich anzumelden. Diese Konten sind spezifisch auf die individuellen Gesundheitsziele ausgerichtet, sei es Gewichtsabnahme, Gewichtszunahme oder das Halten des Gewichts mit einer definierten Zielmenge pro Monat.
RESTful APIs werden genutzt, um eine klare Kommunikation zwischen den Controllern und dem Server zu gewährleisten. Entity Framework Core erleichtert den Zugriff auf die Datenbank, während GitHub Actions für automatisierte Tests, Builds und Bereitstellungen eingesetzt werden.
Das Projektmanagement erfolgt über JIRA, um die Organisation von Aufgaben und Sprints zu ermöglichen. 
Es werden bewährte Entwurfsmuster wie das MVC-Pattern, das Repository Pattern und Dependency Injection angewendet, um eine skalierbare und wartbare Anwendung zu gewährleisten.

UtilityTree:
<img width="554" alt="UtilityTree" src="https://github.com/Chrissi-Ruege/HealthKeeper/assets/123247152/4a0ae9f1-7df7-4da4-83bc-d16fcf926841">

**Comment by ghost** [comment]
Euer Konzept und Aufbau sehen sehr strukturiert und gut durchdacht aus.
Uns ist zudem aufgefallen, dass der Utility Tree sehr ausführlich bearbeitet wurde, da ihr mehr Szenarien habt als gefordert, sehr gut!
Habt ihr jedoch auch noch einen Blogpost für die Aufgaben dieser Woche (9)?
